src/crimson/os/seastore/transaction.h

```diff
@@ -188,15 +188,16 @@ class Transaction {
 
   void mark_delayed_extent_inline(LogicalCachedExtentRef& ref) {
     write_set.erase(*ref);
-    ref->set_paddr(make_record_relative_paddr(offset));
+    ref->set_paddr(make_record_relative_paddr(offset),
+		   /* need_update_mapping: */ false);
```

Comment:
is it `true`?

Reply:
The prior_poffset is only required when invoking `update_mapping` for ool extents

```cpp
  update_mapping_ret update_mapping(
    Transaction& t,
    laddr_t laddr,
    paddr_t prev_addr,
    paddr_t paddr) final;
```

The paddr of inline extent can be set directly in `Cache::complete_commit` without prior_poffset:

https://github.com/ceph/ceph/blob/d5045f7142cb6edf496502ba85b25da78e249bf9/src/crimson/os/seastore/cache.cc#L1433-L1436

Inline extent with prior_poffset seems not affect this process, so it is set `true` to disable keeping the old paddr.

Reply:
I don't see any difference for the delayed paddr mechanism with regard to whether the extent is decided delayed-inline or delayed-ool.

The delayed paddr will be add to the lba tree upon allocation: https://github.com/ceph/ceph/blob/d5045f7142cb6edf496502ba85b25da78e249bf9/src/crimson/os/seastore/transaction_manager.h#L319-L329

And that delayed addr in the lba tree needs to be updated by the meaningful paddr after it is available -- record-relative for inline or absolute for ool. LBA tree should not actually persist the meaningless delayed paddr upon commit.

Reply:
Or probably just drop the delayed-inline logic as there is no way to actually verify this path yet?

Reply:
> LBA tree should not actually persist the meaningless delayed paddr upon commit.

I get it, you are right. When the delayed extents become inline, it also needs to invoke update_mappings to set the paddr record relative.

> Or probably just drop the delayed-inline logic as there is no way to actually verify this path yet?

If it's not necessary to change a delayed extent to an inline extent, dropping it is the better choice.